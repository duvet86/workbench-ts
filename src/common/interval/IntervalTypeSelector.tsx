import React, { ChangeEvent, SFC } from "react";

import {
  createStyles,
  Theme,
  withStyles,
  WithStyles
} from "@material-ui/core/styles";

import { IIntervalTypesDtc } from "common/interval/types";

import FormControl from "@material-ui/core/FormControl";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Select from "@material-ui/core/Select";

interface IProps extends WithStyles<typeof styles> {
  options: IIntervalTypesDtc[];
  value: string;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
}

const styles = ({ spacing: { unit } }: Theme) =>
  createStyles({
    container: {
      flexGrow: 1,
      margin: unit
    }
  });

const IntervalTypeSelector: SFC<IProps> = ({
  classes,
  options,
  value,
  onChange
}) => (
  <FormControl className={classes.container}>
    <InputLabel htmlFor="interval">Interval</InputLabel>
    <Select
      value={value}
      onChange={onChange}
      inputProps={{
        name: "interval"
      }}
    >
      {options.map(({ IntervalType, Label }) => (
        <MenuItem key={IntervalType} value={IntervalType}>
          {Label}
        </MenuItem>
      ))}
    </Select>
  </FormControl>
);

export default withStyles(styles)(IntervalTypeSelector);
