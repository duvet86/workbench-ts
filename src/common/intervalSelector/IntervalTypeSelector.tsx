import React, { ChangeEvent, SFC } from "react";

import { IInterval, IIntervalTypesDtc } from "common/intervalSelector/types";

import FormControl from "@material-ui/core/FormControl";
import Input from "@material-ui/core/Input";
import InputLabel from "@material-ui/core/InputLabel";
import MenuItem from "@material-ui/core/MenuItem";
import Select from "@material-ui/core/Select";

// interface IBackDropProps {
//   onClick: (event: MouseEvent<HTMLDivElement>) => void;
// }

// const CustomBackdropComponent: StatelessComponent<IBackDropProps> = ({
//   onClick
// }) => (
//   <div
//     style={{
//       height: "100%",
//       left: 0,
//       position: "fixed",
//       top: 0,
//       width: "100%",
//       zIndex: -1
//     }}
//     onClick={onClick}
//     aria-hidden="true"
//   />
// );

interface IIntervalTypeProps {
  className?: string;
  intervalTypes: IIntervalTypesDtc[];
  interval: IInterval;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
}

const IntervalTypeSelector: SFC<IIntervalTypeProps> = ({
  className,
  intervalTypes,
  interval,
  onChange
}) => (
  <FormControl className={className}>
    <InputLabel htmlFor="interval">Interval</InputLabel>
    <Select
      // MenuProps={{
      //   BackdropComponent: CustomBackdropComponent
      // }}
      value={interval.type}
      onChange={onChange}
      input={<Input name="interval" id="interval" />}
    >
      {intervalTypes.map(({ IntervalType, Label }) => (
        <MenuItem key={IntervalType} value={IntervalType}>
          {Label}
        </MenuItem>
      ))}
    </Select>
  </FormControl>
);

export default IntervalTypeSelector;
