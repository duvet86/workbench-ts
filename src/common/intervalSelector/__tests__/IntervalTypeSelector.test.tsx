import React from "react";
import { create } from "react-test-renderer";

import IntervalTypeSelector from "common/intervalSelector/IntervalTypeSelector";

test("Test", () => {
  const intervalTypes = [
    {
      IntervalType: "IntervalType",
      Label: "Label",
      SmartIntervals: [{ Key: "SmartIntervals" }]
    }
  ];
  const interval = {
    type: "type",
    string: "string"
  };
  const onChange = () => {
    return;
  };

  const component = create(
    <IntervalTypeSelector
      intervalTypes={intervalTypes}
      interval={interval}
      onChange={onChange}
    />
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
