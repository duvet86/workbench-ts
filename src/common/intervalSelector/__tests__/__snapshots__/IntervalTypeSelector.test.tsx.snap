// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test 1`] = `
<div
  className="MuiFormControl-root-01"
>
  <label
    className="MuiFormLabel-root-010 MuiFormLabel-filled-014 MuiInputLabel-root-05 MuiInputLabel-formControl-06 MuiInputLabel-animated-09 MuiInputLabel-shrink-08"
    data-shrink={true}
    htmlFor="interval"
  >
    Interval
  </label>
  <div
    className="MuiInput-root-022 MuiInput-formControl-023 MuiInput-underline-026"
  >
    <div
      className="MuiSelect-root-017"
    >
      <div
        aria-haspopup="true"
        aria-owns={null}
        aria-pressed="false"
        className="MuiSelect-select-018 MuiSelect-selectMenu-019 MuiInput-input-030"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        id="interval"
        name="interval"
        type="hidden"
        value="type"
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-035 MuiSelect-icon-021"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</div>
`;
